File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int alpha[257];
        memset(alpha , 0 , sizeof(alpha));

        int size = s.size() , low = 0 ,high = 0 , res = 0;

        while(high < size){
            alpha[s[high]]++;

            while(alpha[s[high]] > 1){
                alpha[s[low]]--;
                low++;
            }
            res = max(res , high - low + 1);
            high++;
        }

        return res;
    }
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        st = set()
        (low, high , size) = (0,0 , len(s))
        res = 0

        while high < size:
            while s[high] in st:
                st.remove(s[low])
                low += 1
            st.add(s[high])
            res = max(res, high - low+1)
            high+=1

        return res
        